spec/std/va_list_spec.cr

```diff
@@ -0,0 +1,37 @@
+require "../spec_helper"
+
+describe VaList do
+  it "works with C code" do
```

Comment:
CI passes? No clue beyond that.

Reply:
Well, compiler CI is not enabled yet, but I tried and it works :>

Reply:
This is a stdlib spec btw.

Reply:
Wait, *why* is it a stdlib spec? All such specs and usages of `test_c` so far are under _spec/compiler_. Also I think std specs can't require the top-level spec helper (which seems to be compiler spec helper...)

As-is, I'm pretty sure we're back to https://github.com/crystal-lang/crystal/pull/9351#issuecomment-633611186

> This PR as-is makes the stdlib spec suite require the compiler source.

\- and that's a big deal.

And in that case, additionally, this PR forgets to add the file to _spec/win32_std_spec.cr_ (probably commented-out because it wouldn't work for the prior reason)

Reply:
Well, it's not testing any compiler functionality though :/

Reply:
@jhass Hmm it's a bit strange to duplicate the `test_c` function in another place, which seems to be what you're doing. Also getting it to support Windows within the scope of this PR is pretty hopeless. That'll be https://github.com/crystal-lang/crystal/pull/9348/files#diff-906deee07fd614ba421ecd7a4a66ffd1, which you seem to be re-implementing little by little.

There are some things to consider here.
1. If it's really a stdlib-only spec, then why are we relying on invoking the compiler?
2. Maybe it's not worth the fight, just put it into compiler specs?

Reply:
There's nothing compiler dependent here. It's invoking the compiler because the alternative is to setup something like `spec/ext` for the stdlib specs, which I felt was the path of higher resistance.

Reply:
Well, sorry about that confusion, at first I thought it'd be handled seamlessly by just adding it to compiler specs (would be fully compatible and working with #9348 regardless of merge order) and I hadn't realized about the dependency on the compiler (which is definitely a blocker, I think). If you add it to std specs, that's a bigger mess. You can add a commented out `require` line in win_std_spec.cr, I suppose...

Reply:
If we feel the need to duplicate these helpers, the right solution would be to move them into a commonly accessible location, which is under _spec/support/_. And I suppose a general wish I'd have (I mean beyond this PR) is to requite those explicitly as needed rather than clumping everything into one *spec_helper*.

Reply:
I think having such a helper in the std specs could prove useful beyond this.

By the way, I see `flag?(:win32)` and `flag?(:windows)` used seemingly at random, what's the rationale between using either?

Reply:
Well `win32` is explicitly defined

https://github.com/crystal-lang/crystal/blob/edc261c85bc198abedb33f90f04d9b091337bdfb/src/compiler/crystal/semantic/flags.cr#L40

and is more widespread;
while `windows` seems to be mostly in older code, and is available just as a side effect of splitting the target triple.
So I'd say just use `win32`.

Reply:
But we rely on the "side effect of splitting the target triple" everywhere else, it's like, an intended feature?!

Reply:
Maybe true but best to just follow the convention: `win32` ~110 occurences vs `windows` ~17 occurences

Reply:
Agh I see you're running into too many subtleties of testing on Windows which are all pending in #9348. Please can you just move the helper without adding Windows handling. The addition to _spec/win32_std_spec.cr_ can be commented out.

Reply:
Yeah, I feel like this should work, why the hell wouldn't it find the source file?

Reply:
I'm pretty sure it's not finding bin/crystal. I'm solving those cases in #9348 as well: https://github.com/crystal-lang/crystal/pull/9348/commits/902b13d431fee9e41977f4704f40f72703bc913f

Reply:
Oh well, there you go.

Reply:
Just has a leftover `temp_c.#{obj_ext}` (with `obj_ext` not defined) now

Reply:
Good catch, thanks. I guess I'm tired...